File: src/main/java/ru/xw1w1/showdamage/commands/ShowDamageCommand.java
package ru.xw1w1.showdamage.commands;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import ru.xw1w1.showdamage.Main;
import ru.xw1w1.showdamage.utils.ShowDamage;
import ru.xw1w1.showdamage.utils.TextUtils;

public class ShowDamageCommand extends TextUtils implements CommandExecutor {

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (args.length == 0) {
            sender.sendMessage(red("Usage:  /showdamage debugdamage|reloadconfig"));
        }
        if (args.length >= 1) {
            if (args[0].equals("debugdamage")) {
                ShowDamage showDamage = new ShowDamage();
                showDamage.show("12.25", ((Player) sender).getLocation(), false);
            }
        }
        return true;
    }
}

File: src/main/kotlin/dev/craftwarestudios/showdamage/command/ShowDamageCommand.kt
package dev.craftwarestudios.showdamage.command

import dev.craftwarestudios.showdamage.ShowDamage
import dev.craftwarestudios.showdamage.util.TextUtils.green
import org.bukkit.NamespacedKey
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.bukkit.persistence.PersistentDataType
import org.incendo.cloud.paper.LegacyPaperCommandManager

class ShowDamageCommand(manager: LegacyPaperCommandManager<CommandSender>) {
    init {
        val commandBuilder = manager.commandBuilder("showdamage", "sd")

        manager.command(commandBuilder
            .literal("reload-config")
            .permission(PERMISSION_RELOAD)
            .handler { ctx ->
                ShowDamage.instance.reloadConfiguration()
                ctx.sender().sendMessage(green("Config reloaded"))
            }
        )

        manager.command(commandBuilder
            .literal("disable-popup")
            .permission(PERMISSION_DISABLE_POPUP)
            .senderType(Player::class.java)
            .handler { ctx ->
                val container = ctx.sender().persistentDataContainer
                val current = container.get(PDC_VISIBILITY_KEY, PersistentDataType.INTEGER) ?: 1
                val toggled = -current

                container.set(PDC_VISIBILITY_KEY, PersistentDataType.INTEGER, toggled)
                ctx.sender().sendMessage(green("Visibility toggled to ", toggled))
            }
        )
    }

    companion object {
        const val PERMISSION_RELOAD = "showdamage.command.reloadconfig"
        const val PERMISSION_DISABLE_POPUP = "showdamage.command.disablepopup"

        val PDC_VISIBILITY_KEY = NamespacedKey(ShowDamage.instance, "showdamage.damage.visible")
    }
}